src/studio/src/designer/frontend/packages/schema-editor/src/components/Editor.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { ArrowDropDown, ArrowRight, ArchiveOutlined } from '@material-ui/icons';
+import { TabContext, TabList, TabPanel, TreeItem, TreeView } from '@material-ui/lab';
+import { useSelector, useDispatch } from 'react-redux';
+import { AppBar, Button } from '@material-ui/core';
+import { ILanguage, ISchema, ISchemaState, UiSchemaItem } from '../types';
+import { setUiSchema, setJsonSchema, updateJsonSchema, addRootItem, setSchemaName } from '../features/editor/schemaEditorSlice';
+import SchemaItem from './SchemaItem';
+import { getDomFriendlyID, getTranslation } from '../utils';
+import SchemaInspector from './SchemaInspector';
+import { SchemaTab } from './SchemaTab';
+
+const useStyles = makeStyles({
+  root: {
+    height: '100%',
+    width: '100%',
+    display: 'inline-flex',
+    flexWrap: 'wrap',
+    '& > main': {
+      flex: 1,
+      maxWidth: 'calc(100% - 501px)',
+    },
+    '& > aside': {
+      position: 'sticky',
+      top: 110,
+      width: 500,
+      height: 'calc(100vh - 110px)',
+      overflowX: 'clip',
+      overflowY: 'auto',
+    },
+  },
+  editor: {
+    backgroundColor: 'white',
+    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
+    minHeight: 200,
+    margin: 18,
+  },
+  toolbar: {
+    display: 'flex',
+    background: '#fff',
+    padding: 8,
+    boxShadow: '0px 4px 4px rgba(0, 0, 0, 0.25)',
+    '& > button': {
+      margin: 4,
+      background: '#fff',
+      '&:last-child': {
+        marginLeft: 'auto',
+      },
+    },
+  },
+  appBar: {
+    border: 'none',
+    boxShadow: 'none',
+    backgroundColor: '#fff',
+    color: '#000',
+    '& .Mui-Selected': {
+      color: '#6A6A6A',
+    },
+    '& .MuiTabs-indicator': {
+      backgroundColor: '#006BD8',
+    },
+  },
+  tab: {
+    minWidth: 70,
+  },
+  treeItem: {
+    marginLeft: 8,
+    '&.Mui-selected': {
+      background: '#E3F7FF',
+      border: '1px solid #006BD8',
+      boxSizing: 'border-box',
+      borderRadius: '5px',
+    },
+    '&.Mui-selected > .MuiTreeItem-content .MuiTreeItem-label, .MuiTreeItem-root.Mui-selected:focus > .MuiTreeItem-content .MuiTreeItem-label': {
```

Comment:
It could be. I will have to look at it when I come back. `Editor.tsx` isn't really a new file, most of it is a mv of `schemaEditor.tsx`.
Added some stuff in the interfaces and moved some stuff from `shared/.../DataModelling.tsx` into it.